### ISSUE TITLE
There is any plan to implement Hash like (#!) in this library?

### ISSUE THREAD

**Issue by tufik2** [krasimir/navigo#256]

**Comment by krasimir** [comment]
Hey @tufik2,

Navigo has always been in favor of the people using it. I mean from the early days I'm trying to satisfy most of the requests that come here. I decided to remove the hash-based routing so I have a simple implementation. However, now when we have 100+ unit tests I think it's safe to bring it back. I'll work on it these days and will post here.

Can you share what's your use case and why you can't use the default history-api approach.

**Comment by tufik2** [comment]
Usually we config the server to resolve all URL in a single entry point ://sire.com/ or ://site.com/category/10 and if you reload the app that work well but somecases in some servers we don't havethe capability to re-config server rules so the faster way and a good way to solve it is configuring the router with a hash and that will solve the reload common issue. Thanks.

**Comment by krasimir** [comment]
@tufik2 just released 8.1.0 that supports hash based routing.

**Comment by krasimir** [comment]
Just for the record. I did an example [here](https://github.com/krasimir/navigo/blob/master/examples/hash-mixed/page.html) which is using mixed type of routing. That's quite interesting because it allows us to handle routes but also use anchors in the page:

```
router
  .on("/about", (match) => {
    render("About");
  })
  .on("/products", (match) => {
    render("Products " + JSON.stringify(match.params));
  })
  .on("/login", (match) => {
    render("Login");
  })
  .on("/footer", () => {
    console.log("FOOOOOOOOOTER!");
  })
  .on((match) => {
    render("home");
  })
  .resolve();
```

with the following HTML

```html
<div>
  <a href="/about" data-navigo>About</a>
  <a href="/products?a=b" data-navigo>Products 1</a>
  <a href="/products?c=d" data-navigo>Products 2</a>
  <a href="/login" data-navigo>Login</a>
  <a
    href="/footer"
    data-navigo
    data-navigo-options="resolveOptionsHash:true"
    >Footer</a
  >
</div>
....<lots of stuff here....
<a id="/footer">footer here</a>
```

Clicking on `Footer` makes the URL ending on `#/footer`, the browser scrolls to the anchor and the library resolves the `/footer` handler.
